plugins/temporal/rpc.go

```diff
@@ -0,0 +1,492 @@
+// activity Execute method can return activity.ErrResultPending to
+// indicate the activity is not completed when it's Execute method returns. In that case, this CompleteActivityById() method
+// should be called when that activity is completed with the actual result and error. If err is nil, activity task
+// completed event will be reported; if err is CanceledError, activity task canceled event will be reported; otherwise,
+// activity task failed event will be reported.
+// An activity implementation should use activityID provided in ActivityOption to use for completion.
+// namespace name, workflowID, activityID are required, runID is optional.
+// The errors it can return:
+//  - ErrorWithDetails
+//  - TimeoutError
+//  - CanceledError
+func (r *Rpc) CompleteActivityByID(in CompleteActivityByIdIn, out *CompleteActivityByIdOut) error {
+	ctx := context.Background()
+	var res interface{}
+	var err error
+
+	if in.Err != "" {
+		// complete by id with error
+		err = r.client.CompleteActivityByID(ctx, in.Namespace, in.WorkflowId, in.WorkflowRunId, in.ActivityId, &res, errors.New(in.Err))
+		if err != nil {
+			return err
+		}
+		(*out).Result = res
+		return nil
+	}
+
+	// complete without error
+	err = r.client.CompleteActivityByID(ctx, in.Namespace, in.WorkflowId, in.WorkflowRunId, in.ActivityId, &res, nil)
+	if err != nil {
+		return err
+	}
+	(*out).Result = res
+	return nil
+
+}
+
+type QueryWorkflowIn struct {
+	WorkflowId    string        `json:"wid"`
+	WorkflowRunId string        `json:"rid"`
+	QueryType     string        `json:"query_type"`
+	Args          []interface{} `json:"args"`
+}
+
+// QueryWorkflow queries a given workflow's last execution and returns the query result synchronously. Parameter workflowID
+// and queryType are required, other parameters are optional. The workflowID and runID (optional) identify the
+// target workflow execution that this query will be send to. If runID is not specified (empty string), server will
+// use the currently running execution of that workflowID. The queryType specifies the type of query you want to
+// run. By default, temporal supports "__stack_trace" as a standard query type, which will return string value
+// representing the call stack of the target workflow. The target workflow could also setup different query handler
+// to handle custom query types.
+// See comments at workflow.SetQueryHandler(ctx Context, queryType string, handler interface{}) for more details
+// on how to setup query handler within the target workflow.
+// - workflowID is required.
+// - runID can be default(empty string). if empty string then it will pick the running execution of that workflow ID.
+// - queryType is the type of the query.
+// - args... are the optional query parameters.
+// The errors it can return:
+//  - BadRequestError
+//  - InternalServiceError
+//  - EntityNotExistError
+//  - QueryFailError
+func (r *Rpc) QueryWorkflow(in QueryWorkflowIn, out *payload.RRPayload) error {
```

Comment:
need to double check that this payload properly represents all responses

Reply:
Agree, will be covered with tests